src/utils.jl

```diff
@@ -0,0 +1,246 @@
+# pre-1.11 compat
+if VERSION < v"1.11"
+    function _in!(x, s::Set)
+        xT = convert(eltype(s), x)
+        idx, sh = Base.ht_keyindex2_shorthash!(s.dict, xT)
+        idx > 0 && return true
+        Base._setindex!(s.dict, nothing, xT, -idx, sh)
+        return false
+    end
+else
+    const _in! = Base.in!
+end
+
+# hand-rolled scoped enum
+module JSONTypes
+    primitive type T 8 end
```

Comment:
Using `primitive type` instead of just `struct` here seems ugly and unnecessary.